meson.build

```diff
@@ -1636,7 +1636,12 @@ if get_option('efi')
 else
         have = false
 endif
+
+subdir('src/fundamental')
+subdir('src/boot/efi')
```

Comment:
hmm, why move this up? is that technically required for meson to do the right thing?

Reply:
Yes, `have_gnu_efi` variable is established by `src/boot/efi/meson.build` and that in turn depends on stuff from `src/fundamental/meson.build`.

Reply:
but maybe HAVE_GNU_EFI should then be set in src/boot/efi/meson.build too, and not here?

Reply:
HAVE_GNU_EFI can't be set in `src/boot/efi/meson.build` because configuration data object is used before that subdir is executed in the original order. And variables can't be added to configuration object from which we've already generated `config.h`.